File: Difference_EvenOddDigits/Calculate.py
"""
    Author : Pakshal Jain
    Date : 20/10/2022
    Program : To Calculate Differece Between Summation of Even and Odd Digits of Entered Number
"""

#Function To Calcualate Difference
def Difference(No) :
    Digits = 0;
    Sum = 0;
    Sum2 = 0;
    Diff = 0;

    #Logic
    while(No != 0) :
        Digits = No % 10;
        
        if(Digits % 2 == 0) :
            Sum = Sum + Digits;
        elif(Digits % 2 != 0) :
            Sum2 = Sum2 + Digits;
        
        No = No // 10;

    Diff = Sum - Sum2;

    return Diff;

#Main Function
def main() :
    No = 0;
    iret = 0;

    print("Jay Ganesh...");

    #Taking Input From User
    print("Enter One Number");
    No = int(input());

    #Call To Difference Function
    iret = Difference(No);

    #Printing Difference Calculated
    print("Difference Between Summation of Even And Odd Digits is :",iret);

#Call Back Method To Call Main Function
if __name__ == "__main__" :
    #Call To Main Function
    main();

File: Difference_EvenOddDigits/Calculate.js
/*
    Auhtor : Pakshal Jain
    Date : 20/10/2022
    Program : To Calculate Difference Between Summation of Even and odd Digits of Entered Number
*/
function Difference(No) {
    var Diff = 0;
    var Sum = 0;
    var Sum2 = 0;
    var Digits = 0;
    while (No != 0) {
        Digits = No % 10;
        if (Digits % 2 == 0) {
            Sum = Sum + Digits;
        }
        else if (Digits % 2 != 0) {
            Sum2 = Sum2 + Digits;
        }
        No = Math.floor(No / 10);
    }
    Diff = Sum - Sum2;
    return Diff;
}
function main() {
    var no = 2395;
    var iret = 0;
    console.log("Jay Ganesh...");
    iret = Difference(no);
    console.log("Difference Between Summation of Even and Odd Digits is : %d", iret);
}
main();
